File: test-unit.js
const { getAllGenres, getGenreById, createGenre, updateGenre, deleteGenre } = require('./GenreController');
const GenreService = require('./GenreService');

jest.mock('./GenreService');

describe('GenreController', () => {
    test('getAllGenres', () => {
        // Mocking
        const genres = [new Genre(), new Genre()];
        GenreService.prototype.getAll.mockReturnValue(genres);

        // Test
        const response = getAllGenres();

        // Verificación
        expect(response.status).toBe(200);
        expect(response.body).toEqual(genres);
    });

});

File: test-unit.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import java.util.Arrays;
import java.util.List;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.*;

@ExtendWith(MockitoExtension.class)
public class GenreControllerTest {

    @Mock
    private IGenreService genreService;

    @InjectMocks
    private GenreController genreController;

    @Test
    public void testGetAllGenres() {
        // Mocking
        List<Genre> genres = Arrays.asList(new Genre(), new Genre());
        when(genreService.getAll()).thenReturn(genres);

        // Test
        ResponseEntity<List<Genre>> response = genreController.getAllGenres();

        // Verificación
        assertEquals(HttpStatus.OK, response.getStatusCode());
        assertEquals(genres, response.getBody());
    }

}
